_posts/blog/2022-12-1-vector-library-vs-vector-database.md

```diff
@@ -0,0 +1,108 @@
+
+Weaviate also has a bunch of example use cases on [GitHub](https://github.com/semi-technologies/weaviate-examples){:target="_blank"}. Find your favorite example, give it a star, and try to recreate it yourself!
+
+## Comparison of Features
+The table below summarizes the differences between vector libraries and databases. This is by no means an exhaustive list of features, and not every library or database has the same features. 
+
+|**Feature**|**Vector Library**|**Vector Database** (Weaviate as an example)|
+| Filtering (in combination with Vector Search)| No| Yes|
+| Updatability (CRUD)| No (some do, e.g. hnswlib)| Yes|
+| Incremental importing, concurrent reading while importing| No (some do, e.g. hnswlib) | Yes |
+| Stores objects and vectors | No | Yes|
+|Speed | Typically faster than full-blown database|  Typically slower than pure library|
+| Performance optimized for | In-memory similarity search| End2end-callstack, including: vector search, object retrieval from persistent storage, optional inverted index filtering, network requests, etc. |
+| Durability, crash recovery | No | Yes |
+| Persistence | Only at explicit snapshot | Immediate (after each insert, update or delete) |
+| Sharding (i.e. distribute dataset among more than one machine, automatically merge/aggregate results at query time across machines) | No (there are some 3rd-party projects that wrap sharding around existing libraries) | Yes |
+| Replication | No | Yes (from v1.17) |
+| Automated backups (i.e. to and from cloud storage) | No | Yes |
+| Deployment ecosystem (Docker, K8s, Helm, SaaS) | No (you have to build it yourself) |Yes |
+|SDKs / language clients | No (although most have python bindings) | Yes (Python, Java, JS, Golang) |
+| Execution | Embedded (can be turned into standalone service if you build a simple wrapper app around it) | Standalone service |
+|Communication with app|Foreign Function Interface (FFI) / Language Bindings | Network calls (HTTP, gRPC, etc.) |
+| Multi-tenancy | No | Yes |
+| Arbitrary aggregations of stored objects | No | Yes |
+| Module ecosystem (e.g. vectorization, QnA) | No (have to build it yourself) | Yes |
+| Hybrid BM25+dense vector search | No | Yes (from v1.17) |
+
+## Resources 
+If you’re interested in learning more about vector databases and vector libraries, check out the resources below:
+
+* See how [Weaviate implemented red-black trees](/blog/2022/09/Weaviate-release-1-15.html#faster-imports-for-ordered-data){:target="_blank"} in the v1.15 release. 
+
+* Listen to this [podcast](https://www.youtube.com/watch?v=5o1YTp1IL5o){:target="_blank"} with Meta AI Scientist [Matthijs Douze](https://www.linkedin.com/in/matthijs-douze-750980177/){:target="_blank"} and Abdel Rodriguez, Etienne Dilocker, and Connor Shorten from Weaviate! They talk about Facebook Faiss, product quantization for ANN search, and more. 
```

Comment:
Red-black tree angle: You're right, this link isn't supposed to be included. It was more for the first draft since it discussed HNSW and incremental updates. 

Podcast: This is a good resource for people wanting to learn more about Faiss and product quantization.

Reply:
The podcast is very long though, and has no visuals. There are [shorter guides to Faiss](https://towardsdatascience.com/facebook-ai-similarity-search-7c564daee9eb) and [quantization](https://towardsdatascience.com/understanding-faiss-619bb6db2d1a), though not our content.